File: Algorithms/20. Valid Parentheses/javascript/solution.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
  const stack = [];
  const mapping = {
    "(": ")",
    "{": "}",
    "[": "]"
  };

  for (const char of s) {
    if (char in mapping) {
      stack.push(char);
    } else if (mapping[stack.pop()] !== char) {
      return false;
    }
  }
  return stack.length > 0 ? false : true;
};

File: Algorithms/20. Valid Parentheses/kotlin/solution.kt
import java.util.ArrayDeque

class Solution {
    fun isValid(s: String): Boolean {
        val stack = ArrayDeque<Char>()
        for (char in s) {
            if (char == '(' || char == '{' || char == '[') {
                stack.push(char)
            } else if ((char == ')' && if (stack.contains('(')) stack.pop() != '(' else true)
                || (char == '}' && if (stack.contains('{')) stack.pop() != '{' else true)
                || (char == ']' && if (stack.contains('[')) stack.pop() != '[' else true)
            ) {
                return false
            }
        }
        return stack.isEmpty()
    }
}